gateway/src/config.rs

```diff
@@ -0,0 +1,535 @@
+    pub fn gateway_url(&self) -> Option<&str> {
+        self.gateway_url.as_deref()
+    }
+
+    /// Immutable reference to the identification properties the shard will use.
+    pub const fn identify_properties(&self) -> Option<&IdentifyProperties> {
+        self.identify_properties.as_ref()
+    }
+
+    /// Intents that the shard requests when identifying with the gateway.
+    pub const fn intents(&self) -> Intents {
+        self.intents
+    }
+
+    /// Maximum threshold at which point the gateway will stop sending a guild's
+    /// member list in Guild Create events.
+    pub const fn large_threshold(&self) -> u64 {
+        self.large_threshold
+    }
+
+    /// Immutable reference to the queue in use by the shard.
+    pub fn queue(&self) -> &Arc<dyn Queue> {
+        &self.queue
+    }
+
+    /// Return an immutable reference to the presence to set when identifying
+    /// with the gateway.
+    ///
+    /// This will be the bot's presence. For example, setting the online status
+    /// to Do Not Disturb will show the status in the bot's presence.
+    pub const fn presence(&self) -> Option<&UpdatePresencePayload> {
+        self.presence.as_ref()
+    }
+
+    /// Whether [outgoing message] ratelimiting is enabled.
+    ///
+    /// [outgoing message]: crate::Shard::send
+    pub const fn ratelimit_messages(&self) -> bool {
+        self.ratelimit_messages
+    }
+
+    /// Session information to resume a shard on initialization.
+    pub const fn session(&self) -> Option<&Session> {
+        self.session.as_ref()
+    }
+
+    /// Immutable reference to the TLS connector in use by the shard.
+    pub(crate) const fn tls(&self) -> &TlsContainer {
+        &self.tls
+    }
+
+    /// Immutable reference to the token used to authenticate when identifying
+    /// with the gateway.
+    pub const fn token(&self) -> &str {
+        &self.token
+    }
+}
+
+/// Builder to customize the operation of a shard.
+#[derive(Debug)]
+#[must_use = "builder must be completed to be used"]
+pub struct ConfigBuilder {
+    /// Inner configuration being modified.
+    inner: Config,
+}
+
+impl ConfigBuilder {
+    /// Create a new builder to configure and construct a shard.
+    ///
+    /// Refer to each method to learn their default values.
+    pub fn new(mut token: String, intents: Intents) -> Self {
+        if !token.starts_with("Bot ") {
+            token.insert_str(0, "Bot ");
+        }
+
+        Self {
+            inner: Config {
+                event_types: EventTypeFlags::default(),
+                gateway_url: None,
+                identify_properties: None,
+                intents,
+                large_threshold: 50,
+                presence: None,
+                queue: Arc::new(LocalQueue::new()),
+                ratelimit_messages: true,
+                session: None,
+                tls: TlsContainer::new().expect("failed to build tls"),
```

Comment:
`expect()` should note why an error should not happen. I think this should be replaced by an unwrap and an accompanying panic documentation section.